compiler/rustc_builtin_macros/src/assert.rs

```diff
@@ -41,7 +38,15 @@ pub fn expand_assert<'cx>(
     });
     let args = P(MacArgs::Delimited(DelimSpan::from_single(sp), MacDelimiter::Parenthesis, tokens));
     let panic_call = MacCall {
-        path: Path::from_ident(Ident::new(sym::panic, sp)),
+        path: Path {
+            span: DUMMY_SP,
+            segments: vec![
+                PathSegment::from_ident(Ident::with_dummy_span(kw::PathRoot)),
+                PathSegment::from_ident(Ident::with_dummy_span(sym::core)),
+                PathSegment::from_ident(Ident::new(sym::panic, sp)),
+            ],
+            tokens: None,
+        },
```

Comment:
I tried to qualify this, but got the error:

```
error[E0433]: failed to resolve: could not find `core` in `{{root}}`

error: aborting due to previous error

For more information about this error, try `rustc --explain E0433`.
error: could not compile `core`
```

It seems like the issue is that expanding as `core` doesn't work inside `core`. How should I do it instead?

Reply:
It should be possible to use something like this for the first path element:
```rust
Ident::new(kw::DollarCrate, cx.with_def_site_ctxt(DUMMY_SP)) // `$crate`
```

Reply:
Built-in macros use a function called `std_path` for paths pointing to the standard library, no need to hand-roll it.